### PR TITLE
Ft signup/login route

### PR DESCRIPTION
**Pull request by Ebunwadi**

### CODE REVIEW

File: controllers/userApiLogic.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  });
+};
+
+export const loginUser = async (req, res) => {
+  const { email, password } = req.body;
+  const user = await pool.query('SELECT * FROM users WHERE email = $1', [email]);
+  if (user.rows.length === 0) {
+    return res.status(401).json({
+      status: 'error',
+      error: 'invalid email or password',
```
Comment (courage173) [inline controllers/userApiLogic.js:53]: this should be invalid email

File: controllers/userApiLogic.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    });
+  }
+
+  const validPassword = await bcrypt.compare(password, user.rows[0].password);
+  if (!validPassword) {
+    return res.status(401).json({
+      status: 'error',
+      error: 'invalid email or password',
```
Comment (courage173) [inline controllers/userApiLogic.js:61]: same here. invalid password

File: middleware/validUserInfo.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+const validInfo = (req, res, next) => {
```
Comment (courage173) [inline middleware/validUserInfo.js:1]: this is not a good validation process. please check out joi validation

File: middleware/admin-auth.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+export default (req, res, next) => {
```
Comment (courage173) [inline middleware/admin-auth.js:1]: also this is not a good way to create a middleware

File: test/user-test.js [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+import server from '../server.js';
+
+// ASSERTION STYLE
+chai.should();
+
+chai.use(chaiHttp);
+
+describe('Admin can create an employee user account and both admin/employee can login', () => {
+  // create-user test
+  describe('POST /api/v1/auth/create-user', () => {
```
Comment (courage173) [inline test/user-test.js:12]: why do we have two describe??

**Review by courage173 — CHANGES_REQUESTED** [review]

File: controllers/userApiLogic.js (hunk 1 above)
Comment (Ebunwadi) [reply controllers/userApiLogic.js:53]: okay. noted

File: controllers/userApiLogic.js (hunk 2 above)
Comment (Ebunwadi) [reply controllers/userApiLogic.js:61]: okay sir

File: middleware/validUserInfo.js (hunk 3 above)
Comment (Ebunwadi) [reply middleware/validUserInfo.js:1]: okay, will check it out.

File: middleware/admin-auth.js (hunk 4 above)
Comment (Ebunwadi) [reply middleware/admin-auth.js:1]: hmmm...i will work on it

File: test/user-test.js (hunk 5 above)
Comment (Ebunwadi) [reply test/user-test.js:12]: i am describing two post request instances; signup and login

**Comment by courage173** [comment]
why is this PR closed

**Comment by Ebunwadi** [comment]
i made a mistake, i renamed the branch earlier on github without updating the changes locally, so when pushing from git, it didnt reflect on this branch

**Pull request closed without merging**